src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccount.fs

```diff
@@ -0,0 +1,69 @@
+﻿module Deciders.BankAccount
+
+open System
+open BankAccountPrimitives
+
+type BankAccountOpened =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      CreatedAt: DateTimeOffset
+      Version: int64 }
+
+type DepositRecorded =
+    { BankAccountId: AccountId
+      Amount: decimal
+      CashierId: CashierId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type CashWithdrawnFromATM =
+    { BankAccountId: AccountId
+      Amount: decimal
+      ATMId: AtmId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type BankAccountClosed =
+    { BankAccountId: AccountId
+      Reason: string
+      ClosedAt: DateTimeOffset
+      Version: int64 }
+
+type Event =
+    | BankAccountOpened of BankAccountOpened
+    | DepositRecorded of DepositRecorded
+    | CashWithdrawnFromATM of CashWithdrawnFromATM
+    | BankAccountClosed of BankAccountClosed
+
+type BankAccount =
+    | NotInitialised
+    | Open of
+        {| Id: AccountId
+           Balance: decimal
+           Version: int64 |}
+    | Closed of {| Id: AccountId; Version: int64 |}
+
+let evolve (state: BankAccount) (event: Event) : BankAccount =
+    match (state, event) with
+    | NotInitialised _, BankAccountOpened opened ->
+        Open
+            {| Id = opened.BankAccountId
+               Balance = 0M
+               Version = 1L |}
+    | Open openAccount, DepositRecorded depositRecorded ->
+        Open
+            {| openAccount with
+                Balance = openAccount.Balance + depositRecorded.Amount
+                Version = depositRecorded.Version |}
+    | Open openAccount, CashWithdrawnFromATM cashWithdrawnFromAtm ->
+        Open
+            {| openAccount with
+                Balance = openAccount.Balance - cashWithdrawnFromAtm.Amount
+                Version = cashWithdrawnFromAtm.Version |}
+    | Open openAccount, BankAccountClosed bankAccountClosed ->
+        Closed
+            {| Id = openAccount.Id
+               Version = bankAccountClosed.Version |}
+    | _ -> state
```

Comment:
Give up total matching is a very bad smell - I always end up refactoring away from it (but its a few steps and there's other things to do)

Reply:
Yeah, I know, eventually, I would probably use the explicit matching, but for those samples, I think that it'll do.